xrpl/utils/txn_parser/utils/order_book_parser.py

```diff
@@ -0,0 +1,217 @@
+                return _derive_currency_amount(new_fields_amount)
+    final_fields = node.get("FinalFields")
+    previous_fields = node.get("PreviousFields")
+    if final_fields is not None and previous_fields is not None:
+        final_fields_amount = final_fields.get(side)
+        previous_fields_amount = previous_fields.get(side)
+        if final_fields_amount is not None and previous_fields_amount is not None:
+            final_amount = _derive_currency_amount(final_fields_amount)
+            previous_amount = _derive_currency_amount(previous_fields_amount)
+            value = _calculate_delta(final_amount, previous_amount)
+            # final_amount is being reused because it makes
+            # it easier to return the changed amount.
+            final_amount["value"] = value
+            # From now on you could consider final_amount as changed_amount.
+            return final_amount
+    return None
+
+
+def _get_quality(
+    taker_gets: CurrencyAmount,
+    taker_pays: CurrencyAmount,
+    book_directory: str,
+) -> str:
+    quality_hex = book_directory[-16:]  # last 16 characters of the BookDirectory
+    taker_gets_currency = taker_gets["currency"]
+    taker_pays_currency = taker_pays["currency"]
+    mantissa = float.fromhex(quality_hex[2:])
+    offset = int(quality_hex[:2], 16) - 100
+    scientific_quality = "{:.16f}e{}".format(mantissa, offset)
+    quality = Decimal(scientific_quality)
+    if taker_gets_currency == "XRP" or taker_pays_currency == "XRP":
+        quality = quality * 1000000
+    return str(quality.normalize())
+
+
+def _get_optional_fields(
+    node: NormalizedNode,
+    field_name: str,
+) -> Optional[Any]:
+    new_fields = node.get("NewFields")
+    final_fields = node.get("FinalFields")
+    if new_fields is not None:
+        return new_fields.get(field_name)
+    if final_fields is not None:
+        return final_fields.get(field_name)
+    return None
+
+
+def _get_fields(
+    node: NormalizedNode,
+    field_name: str,
+) -> Optional[Any]:
+    new_fields = node.get("NewFields")
+    final_fields = node.get("FinalFields")
+    if new_fields is not None:
+        return new_fields.get(field_name)
+    if final_fields is not None:
+        return final_fields.get(field_name)
+    return None
+
+
+def _get_offer_change(node: NormalizedNode) -> Optional[AccountOfferChange]:
+    status = _get_offer_status(node)
+    taker_gets = _get_change_amount(node, status, "TakerGets")
+    taker_pays = _get_change_amount(node, status, "TakerPays")
+    account = _get_fields(node, "Account")
+    sequence = _get_fields(node, "Sequence")
+    book_directory = _get_fields(node, "BookDirectory")
+    # if required fields are None return None
+    if (
+        taker_gets is None
+        or taker_pays is None
+        or account is None
+        or sequence is None
+        or book_directory is None
+    ):
+        return None
+    expiration_time = _get_fields(node, "Expiration")
```

Comment:
nit: There should probably be an extra line between this `return None` and `expiration time...` since they are logically distinct. (This is a more general rule I'd apply throughout - if the code within a function can be separated without impacting each other, using a line break to indicate that can help make it more readable. Kinda like paragraphs :) )